plugins/PostProcessingPlugin/scripts/MultiExtColorMix.py

```diff
@@ -0,0 +1,623 @@
+        # Add a reset if there isn't already one in place----------------------------------------------------------
+        if not reset_present:
+            data[len(data)-1] = M163_reset + "T" + str(resume_extruder) + "\n" + data[len(data)-1]
+        return data
+        
+class ParkAndPurge:
+    def park_script(purge_layer: str, data: str, park_x: str, park_y: str, retract_amt: str) -> str:
+        # Put together the park/purge lines to be inserted----------------------------
+        MyCura = Application.getInstance().getGlobalContainerStack()
+        extruder = MyCura.extruderList
+        if int(purge_layer) < 50:
+            zup = " Z12\n"
+            zdn = " Z-12\n"
+        else:
+            zup = " Z2\n"
+            zdn = " Z-2\n"
+        for index, layer in enumerate(data):
+            if ";LAYER:" + str(purge_layer) + "\n" in layer:
+                prev_layer = data[index-1]
+                lines = prev_layer.split("\n")
+                lines.reverse()
+                xloc = ""
+                yloc = ""            
+                for line in lines:
+                    if line.startswith("G0") or line.startswith("G1") or line.startswith("G2") or line.startswith("G3"):
+                        if " X" in line:
+                          xtemp = line.split("X")[1]
+                          xloc = " X" + str(xtemp.split(" ")[0])
+                        if " Y" in line:
+                          ytemp = line.split("Y")[1]
+                          try:
+                            yloc = " Y" + str(ytemp.split(" ")[0])
+                          except:
+                            yloc = " Y" + str(ytemp)
+                    if xloc != "" and yloc != "":
+                        break
+        park_ret_prime = "\n; Color Mix Purge\nG91\nM83\n"
+        park_ret = "\n; Color Mix Purge\nG91\nM83\n"
+        park_prime = "\n; Color Mix Purge\nG91\nM83\n"
+        park_none = "\n; Color Mix Purge\nG91\nM83\n"
+        firmware_retract = bool(MyCura.getProperty("machine_firmware_retract", "value"))
+        speed_travel = str(int(extruder[0].getProperty("speed_travel", "value")) * 60)
+        speed_print = str(int(extruder[0].getProperty("speed_print", "value")) * 60)
+        speed_z_hop = str(int(extruder[0].getProperty("speed_z_hop", "value")) * 60)
+        retract_distance = str(extruder[0].getProperty("retraction_amount", "value"))
+        speed_retract = " F" + str(int(extruder[0].getProperty("retraction_retract_speed", "value")) *60)
+        relative_ext = bool(MyCura.getProperty("relative_extrusion", "value"))
+        if relative_ext:
+            relative_str = "M83\n"
+        else:
+            relative_str = "M82\n"
+        if firmware_retract:
+            E_retract = "G10\n"
+            E_prime = "G11\n"
+        else:
+            E_retract = "G1" + speed_retract + " E-" + retract_distance + "\n"
+            E_prime = "G1" + speed_retract + " E" + retract_distance + "\n"
```

Comment:
This is a nice example where [string templates](https://peps.python.org/pep-0498/) can be nicely used
```suggestion
            E_retract = f"G1{speed_retract} E-{retract_distance}\n"
            E_prime = f"G1{speed_retract} E{retract_distance}\n"
```